transformers4rec/torch/masking.py

```diff
@@ -338,21 +373,42 @@ def _compute_masked_targets(self, item_ids: torch.Tensor, training=False) -> Mas
         ----------
         item_ids: torch.Tensor
             Sequence of input itemid (target) column
+        {flags_parameters_docstrings}
 
         Returns
         -------
         labels: torch.Tensor
             Sequence of masked item ids.
         mask_labels: torch.Tensor
             Masking schema for masked targets positions.
+        {flags_parameters_docstrings}
+
+        `Note:` During inference, the inputs are extended with one additional
+            [MASK] item embeddings. This position is then used to retrieve
+            the final hidden representation from the transformer block.
+            This is needed to take into account the actual target position
+            when applying the transformer layer.
         """
+        non_padded_mask = item_ids != self.padding_idx
+        rows_ids = torch.arange(item_ids.size(0), dtype=torch.long, device=item_ids.device)
+
+        if not training and not testing:
+            # At inference we extend the input with a [MASK] element at the first padded position
+            # to take into account the positional encoding of the target
+            labels = torch.full(
+                (item_ids.shape[0], item_ids.shape[1] + 1),
+                self.padding_idx,
+                dtype=item_ids.dtype,
+                device=item_ids.device,
+            )
+            last_item_sessions = non_padded_mask.sum(dim=1)
+            labels[rows_ids, last_item_sessions] = item_ids[rows_ids, last_item_sessions - 1]
```

Comment:
Is this inference mode? From `last_item_sessions - 1` it seems we are masking the last item of the sequence, which is expected for evaluation mode, but not for inference mode, right?

Reply:
Yes, it is the inference mode. Here the `labels` are of shape `[bs, max_length+1]` while the input `item_ids` is of shape `[bs, max_length]`. 
The logic here is to set the label at position `last_item_sessions` (which is the position immediately after the last interacted item) with the last interacted item id given by `item_ids[rows_ids, last_item_sessions - 1]`. So that we can define the boolean mask using the check  ```mask_labels = labels != self.padding_idx``` .  So, the resulting mask will have `True` values only for the position `mask[rows_ids, last_item_sessions]`

Reply:
I see @sararb . Makes sense. Thanks